base_layer/core/src/covenants/fields.rs

```diff
@@ -105,7 +105,7 @@ impl OutputField {
                 // sidechain_features because there is no reference to Option<FixedHash> for
                 // sidechain_features.contract_id to return.
                 match &output.features.sidechain_features {
-                    Some(SideChainFeatures { ref contract_id, .. }) => contract_id as &dyn Any,
+                    Some(v) => &v.contract_id as &dyn Any,
```

Comment:
You could keep the pattern matching, and dereference the box in the match:
`match &*output.features.sidechain_features` 

What you have is fine though, unless it is the reason that you added the debug trait bound above?

Reply:
You cant deref Option<Box<T>>
Or at least the compiler does not want to allow me to do it.